src/cli/parser.rs

```diff
@@ -203,12 +283,22 @@ pub fn parse_color(color: &str) -> Option<Color> {
 
     if let Some(caps) = RE_HSL.captures(color) {
         let mh = float_to_f64(caps.get(1).unwrap().as_str());
-        let ms = float_to_f64(caps.get(2).unwrap().as_str());
-        let ml = float_to_f64(caps.get(3).unwrap().as_str());
+        let unit = match caps.get(2) {
+            Some(s) => s.as_str(),
+            None => "deg",
+        };
+        let ms = float_to_f64(caps.get(3).unwrap().as_str());
+        let ml = float_to_f64(caps.get(4).unwrap().as_str());
 
         match (mh, ms, ml) {
             (Some(h), Some(s), Some(l)) => {
-                let h = f64::from(h);
+                let h = match unit {
+                    "°" | "deg" => h,
+                    "grad" => h * (18.0 / 20.0),
+                    "rad" => h.to_degrees().round(), // Better to round it (because of PI being an irrational number)
```

Comment:
I don't think we should round here. That's just a loss of precision.

Reply:
Okay I'll keep it as it.